File: src/main/java/ru/pearx/pmdumper/dumpers/DumperPotions.java
package ru.pearx.pmdumper.dumpers;

import net.minecraft.item.ItemStack;
import net.minecraft.potion.Potion;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.common.registry.ForgeRegistries;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import ru.pearx.pmdumper.dumpers.base.IDumper;
import ru.pearx.pmdumper.dumpers.base.PMDData;

import java.util.*;

/**
 * Created by mrAppleXZ on 14.06.17 11:39.
 */
@SideOnly(Side.CLIENT)
public class DumperPotions implements IDumper
{
    @Override
    public String getName()
    {
        return "potions";
    }

    @Override
    public PMDData getData()
    {
        List<List<String>> lst = new ArrayList<>();
        Map<String, Integer> counts = new HashMap<>();
        lst.add(Arrays.asList("ID", "Name", "Class Name", "Is Bad Effect", "Is Instant", "Is Beneficial", "Curative Items"));

        for(Map.Entry<ResourceLocation, Potion> entr : ForgeRegistries.POTIONS.getEntries())
        {
            ResourceLocation loc = entr.getKey();
            Potion p = entr.getValue();
            StringBuilder curative = new StringBuilder();
            for(ItemStack stack : p.getCurativeItems())
            {
                curative.append(stack.toString());
                curative.append(System.lineSeparator());
            }
            if(p.getCurativeItems().size() > 0)
                curative.delete(curative.length() - System.lineSeparator().length(), curative.length());
            lst.add(Arrays.asList(loc.toString(), p.getName(), p.getClass().getName(), Boolean.toString(p.isBadEffect()), Boolean.toString(p.isInstant()), Boolean.toString(p.isBeneficial()), curative.toString()));
            PMDData.plusCounts(counts, loc.getResourceDomain(), 1);
        }

        return new PMDData(lst, counts);
    }
}

File: src/main/kotlin/ru/pearx/pmdumper/dumper/standard/DumperPotions.kt
@file:JvmMultifileClass
@file:JvmName("StandardDumpers")
package ru.pearx.pmdumper.dumper.standard

import net.minecraft.util.ResourceLocation
import net.minecraft.util.text.translation.I18n
import net.minecraftforge.fml.common.registry.ForgeRegistries
import ru.pearx.pmdumper.ID
import ru.pearx.pmdumper.dumper.dumper
import ru.pearx.pmdumper.utils.appendTo
import ru.pearx.pmdumper.utils.client
import ru.pearx.pmdumper.utils.toHexColorString
import ru.pearx.pmdumper.utils.toPlusMinusString

val DumperPotions = dumper {
    registryName = ResourceLocation(ID, "potions")
    header = listOfNotNull("ID", "Display Name", "Name", "Class Name", "Is Bad Effect", "Is Instant", client("Is Beneficial"), client("Status Icon Index"), client("Liquid Color"), "Curative Items", client("Attribute Modifiers"))
    amounts {
        for (potion in ForgeRegistries.POTIONS)
            this += potion.registryName
    }
    iterator {
        for (potion in ForgeRegistries.POTIONS) {
            with(ArrayList<String>(header.size)) {
                with(potion) {
                    add(registryName.toString())
                    add(I18n.translateToLocalFormatted(name))
                    add(name)
                    add(this::class.java.name)
                    add(isBadEffect.toPlusMinusString())
                    add(isInstant.toPlusMinusString())
                    client {
                        add(isBeneficial.toPlusMinusString())
                        add(statusIconIndex.toString())
                        add(liquidColor.toHexColorString())
                    }
                    add(StringBuilder().apply {
                        var start = true
                        for (item in curativeItems) {
                            if (start)
                                start = false
                            else
                                appendln()
                            item.appendTo(this)
                        }
                    }.toString())
                    client {
                        add(
                            StringBuilder().apply {
                                var start = true
                                for ((attribute, modifier) in attributeModifierMap) {
                                    if (start)
                                        start = true
                                    else
                                        appendln()
                                    append(attribute.name)
                                    append(": ")
                                    append(modifier)
                                }
                            }.toString()
                        )
                    }
                }
                yield(this)
            }
        }
    }
}